### PR TITLE
LP-320 Add service unavailable middleware and page

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-320

### Change description

Add a service unavailable page to be shown when env property SHOW_SERVICE_UNAVAILABLE_PAGE is set to 'true'

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/presentation/test/integration/service-availability.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const SERVICE_UNAVAILABLE_TEXT = "Sorry, the service is unavailable";
+
+describe("Service Availability tests", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const setServiceUnavailable = (value: boolean) => {
+    jest.replaceProperty(configConstants, 'SHOW_SERVICE_UNAVAILABLE_PAGE', value);
+  };
```
Comment (markpit) [inline src/presentation/test/integration/service-availability.test.ts:18]: was unable to get it working with replace process.env, the replace worked but config had already got the process env values before we can change them, so had to replace the property in config.

File: src/views/service-unavailable.njk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{% extends "layout.njk" %}
+
+{% block pageTitle %}
+  Service unavailable - Limited Partnerships - GOV.UK
+{% endblock %}
```
Comment (markpit) [inline src/views/service-unavailable.njk:5]: have to hard code the page title as no locales translation will be available

**Review by lduranteau — APPROVED** [review]

**Pull request merged**